bundles/core/org.eclipse.smarthome.core.thing.test/src/test/groovy/org/eclipse/smarthome/core/thing/internal/ThingManagerOSGiTest.groovy

```diff
@@ -1010,6 +1010,129 @@ class ThingManagerOSGiTest extends OSGiTest {
+        }
+
+        assertThat infoEvent.getType(), is(ThingStatusInfoEvent.TYPE)
+        assertThat infoEvent.getTopic(), is("smarthome/things/binding:type:id/status")
+        assertThat infoEvent.getStatusInfo().getStatus(), is(ThingStatus.INITIALIZING)
+        assertThat infoEvent.getStatusInfo().getStatusDetail(), is(ThingStatusDetail.NONE)
+        assertThat infoEvent.getStatusInfo().getDescription(), is(null)
+
+        assertThat infoChangedEvent.getType(), is(ThingStatusInfoChangedEvent.TYPE)
+        assertThat infoChangedEvent.getTopic(), is("smarthome/things/binding:type:id/statuschanged")
+        assertThat infoChangedEvent.getStatusInfo().getStatus(), is(ThingStatus.INITIALIZING)
+        assertThat infoChangedEvent.getStatusInfo().getStatusDetail(), is(ThingStatusDetail.NONE)
+        assertThat infoChangedEvent.getStatusInfo().getDescription(), is(null)
+        assertThat infoChangedEvent.getOldStatusInfo().getStatus(), is(ThingStatus.UNINITIALIZED)
+        assertThat infoChangedEvent.getOldStatusInfo().getStatusDetail(), is(ThingStatusDetail.NONE)
+        assertThat infoChangedEvent.getOldStatusInfo().getDescription(), is(null)
+
+        infoEvent = null
+        infoChangedEvent = null
+
+        // set status to ONLINE (INITIALIZING -> ONLINE)
+        setDefaultLocale(Locale.ENGLISH)
+
+        ThingStatusInfo statusInfo = ThingStatusInfoBuilder.create(ThingStatus.ONLINE, ThingStatusDetail.NONE).withDescription("@text/online").build()
+        callback.statusUpdated(THING, statusInfo)
+
+        waitForAssert {
+            assertThat infoEvent, not(null)
+            assertThat infoChangedEvent, not(null)
+        }
+
+        assertThat infoEvent.getType(), is(ThingStatusInfoEvent.TYPE)
+        assertThat infoEvent.getTopic(), is("smarthome/things/binding:type:id/status")
+        assertThat infoEvent.getStatusInfo().getStatus(), is(ThingStatus.ONLINE)
+        assertThat infoEvent.getStatusInfo().getStatusDetail(), is(ThingStatusDetail.NONE)
+        assertThat infoEvent.getStatusInfo().getDescription(), is("Thing is online.")
+
+        assertThat infoChangedEvent.getType(), is(ThingStatusInfoChangedEvent.TYPE)
+        assertThat infoChangedEvent.getTopic(), is("smarthome/things/binding:type:id/statuschanged")
+        assertThat infoChangedEvent.getStatusInfo().getStatus(), is(ThingStatus.ONLINE)
+        assertThat infoChangedEvent.getStatusInfo().getStatusDetail(), is(ThingStatusDetail.NONE)
+        assertThat infoChangedEvent.getStatusInfo().getDescription(), is("Thing is online.")
+        assertThat infoChangedEvent.getOldStatusInfo().getStatus(), is(ThingStatus.INITIALIZING)
+        assertThat infoChangedEvent.getOldStatusInfo().getStatusDetail(), is(ThingStatusDetail.NONE)
+        assertThat infoChangedEvent.getOldStatusInfo().getDescription(), is(null)
+
+        infoEvent = null
+        infoChangedEvent = null
+
+        // set status to OFFLINE (ONLINE -> OFFLINE)
```

Comment:
what do you exactly mean by ??? 

The useful comment? If yes I thought it might be best practice in the ```ThingManagerOSGiTest``` as already implemented for the test ```ThingManager posts thing status changed events if the status of a thing is changed```

Do you want me to remove the comment?